crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_map.rs

```diff
@@ -211,16 +159,44 @@ pub(crate) fn unnecessary_map(
     checker.diagnostics.push(diagnostic);
 }
 
+fn is_list_set_or_dict(func: &Expr, semantic: &SemanticModel) -> bool {
+    semantic
+        .resolve_qualified_name(func)
+        .is_some_and(|qualified_name| {
+            matches!(
+                qualified_name.segments(),
+                ["" | "builtins", "list" | "set" | "dict"]
+            )
+        })
+}
+
+fn map_lambda_and_iterables<'a>(
+    call: &'a ast::ExprCall,
+    semantic: &'a SemanticModel,
+) -> Option<(&'a Parameters, &'a Expr, &'a [Expr])> {
+    if !semantic.match_builtin_expr(&call.func, "map") {
+        return None;
+    }
+
```

Comment:
I suggest to remove the lambda here, it reduces the tuple arity and makes it clearer what the second expression is
```suggestion
fn map_lambda_and_iterables<'a>(
    call: &'a ast::ExprCall,
    semantic: &'a SemanticModel,
) -> Option<(&'a ExprLambda, &'a [Expr])> {
```

It also allows to simplify `lambda_has_expected_arity` to `lambda_has_expected_arity(lambda)` and you can move the check if `parameters` is `Some` into `lambda_has_expected_arity`